api/app/utils/pii_detection.py

```
@@ -0,0 +1,249 @@
+    # PII pattern definitions
+    PATTERNS = {
+        "email": r"[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}",
+        "ip_address": r"\b(?:\d{1,3}\.){3}\d{1,3}\b",
+        "bitcoin_legacy": r"\b[13][a-km-zA-HJ-NP-Z1-9]{25,34}\b",
+        "bitcoin_bech32": r"\bbc1[a-z0-9]{39,59}\b",
+        "matrix_id": r"@[a-zA-Z0-9._-]+:[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}",
+        "uuid": r"\b[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}\b",
+        "phone": r"\b(?:\+\d{1,3}[-. ]?)?\(?\d{3}\)?[-. ]?\d{3}[-. ]?\d{4}\b",
+        "api_key": r"\b[a-zA-Z0-9]{32,}\b",
+        "long_number": r"\b\d{8,}\b",
+    }
```

Ruff (0.13.1)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)